pwnlib/util/packing.py

```diff
@@ -538,12 +550,66 @@ def flat(*args, **kwargs):
 
     preprocessor = kwargs.pop('preprocessor', lambda x: None)
     word_size    = kwargs.pop('word_size', None)
+    address      = kwargs.pop('address', 0)
 
     if kwargs != {}:
-        raise TypeError("flat() does not support argument %r" % kwargs.popitem()[0])
+        raise TypeError("flat() got an unexpected keyword argument %r" % kwargs.popitem()[0])
 
-    return _flat(args, preprocessor, make_packer(word_size))
+    for _ in xrange(10000):
+        try:
+            return _flat(args, preprocessor, make_packer(word_size), address)
+        except FitOffsetsChangedException:
+            pass
+    else:
+        raise ValueError("Invalid flat() arguments: %r" % args)
+
+class FitOffsetsChangedException(Exception):
+    pass
 
+class FitLabel(object):
+    """Label mechanism for use with :func:`.packing.fit`
+    """
+    _address = None
+    _contents = None
+
+    def __init__(self, contents=None, address=None):
+        self._contents = contents
+        self._address = address
+
+    def __call__(self, contents):
+        self.contents = contents
+        return self
+
+    def __flat_at__(self, address):
+        self.address = address
+        return self.contents
+
+    def __len__(self):
+        return len(self.contents)
+
+    @property
+    def address(self):
+        return int(self._address or 0)
+
+    @address.setter
+    def address(self, new):
```

Comment:
Should there be a way to "manually" set the address of a label, without causing an exception?

Reply:
Only when constructing the `Label` via `__init__`